jax/api_util.py

```diff
@@ -82,6 +84,37 @@ def argnums_partial(f, dyn_argnums, args):
   dyn_args = tuple(args[i] for i in dyn_argnums)
   return _argnums_partial(f, dyn_argnums, fixed_args), dyn_args
 
+def donation_vector(donate_argnums, args) -> Tuple[bool, ...]:
+  """Returns a tuple with a boolean value for each leaf in args."""
+  res = []
+  for i, arg in enumerate(args):
+    donate = bool(i in donate_argnums)
+    res.extend((donate,) * tree_structure(arg).num_leaves)
+  return tuple(res)
+
+def rebase_donate_argnums(donate_argnums: Iterable[int],
+                          static_argnums: Iterable[int]):
+  """Rebases donate_argnums on static_argnums."""
```

Comment:
It wasn't immediately clear to me what "rebase" meant (I figured it out after a few seconds of thinking), and I think an example would make it clear right away.

Reply:
Have added a comment.